### DISCUSSION TITLE
CTkFrame.place() position not the same as tkinter.Frame.place()

### DISCUSSION THREAD

**Discussion by jh-jensen** [TomSchimansky/CustomTkinter#1167] [Q&A]
Hello. 
I am using .place() to position a frame somewhere on the screen. However, if I give the .place() y attribute a value of, say, 700 for the CTkFrame widget - then it will be placed farther down the screen than if I gave a tkinter.Frame widget the same .place() y value. Everything is the same about the widgets. Not sure what causes this behaviour, but it would be nice with either an explanation or a fix, because I had to revert back to the tkinter.Frame and style it myself for both dark/light theme because the x/y coordinates can't be used for the CTkFrame widget to position it directly beneath another widget, because the w_xinfo and w_yinfo values by the other widget don't match up with where the CTkFrame widget is placed, only the tkinter.Frame widget matches.

I've included a picture of two frames with different bgs and the same .place() values to illustrate the problem: [https://tinypic.host/i/y700.Zy31C](url)

```
self.lsbx_test = customtkinter.CTkFrame(self, fg_color="red", width=300, height=300)
self.lsbx_test.place(anchor="nw", y=700)

self.frm_position_help = tkinter.Frame(self, background="blue", width=300, height=300)
self.frm_position_help.place(anchor="nw", y=700)
```

I normally wouldn't use place anyway, but this specific use kinda requires it.

**Comment by Wolf-SO** [comment]
The CustonTkinter documentation [explicitly advises against using `.place()`](https://github.com/TomSchimansky/CustomTkinter/wiki/App-structure-and-layout). This is not really a restriction because `.place()` (being explicit placement) actually works against any layout.

**Reply by jh-jensen** [reply]
> The CustonTkinter documentation [explicitly advises against using `.place()`](https://github.com/TomSchimansky/CustomTkinter/wiki/App-structure-and-layout). This is not really a restriction because `.place()` (being explicit placement) actually works against any layout.

Yes I know it's advised against, but for certain cases, .place() could be the only option. For example, I had to place an autocomplete listbox widget directly beneath an entry box, but couldn't use grid or pack because that interfered with the opsition of several widgets underneath the entry box. I tried grid/pack_propagate, which keeps the widgets under the entry box in place, but then the listbox cannot grow past the height of the frame the entry box is in. In addition, I wanted the listbox to appear over on top of the widgets beneath the entry widget if the listbox height grew large enough to warrant it. I didn't really see any other option than to use the .place() with .lift() and entrybox.winfo_x/y for this. Is there a better way to do it?

**Comment by Wolf-SO** [comment]
The scenario you describe makes sense. But how would you display an autocomplete widget when there is no more room in the surrounding window (perhaps by enlarging it) or on the screen itself? Wouldn't that be a case for a separate lightweight top-level window, such as one might use for menus or tooltips?
